packages/x-tree-view/src/TreeItem/TreeItem.tsx

```diff
@@ -283,40 +229,33 @@ export const TreeItem = React.forwardRef(function TreeItem(
   function handleFocus(event: React.FocusEvent<HTMLLIElement>) {
     // DOM focus stays on the tree which manages focus with aria-activedescendant
     if (event.target === event.currentTarget) {
-      let rootElement: any;
-
-      if (typeof event.target.getRootNode === 'function') {
-        rootElement = event.target.getRootNode();
-      } else {
-        rootElement = ownerDocument(event.target);
-      }
-
-      rootElement.getElementById(treeId).focus({ preventScroll: true });
+      instance.focusRoot();
     }
 
-    const unfocusable = !disabledItemsFocusable && disabled;
-    if (instance && !focused && event.currentTarget === event.target && !unfocusable) {
+    const canBeFocused = !disabledItemsFocusable && disabled;
```

Comment:
I'm doing some mental gymnastics about his one, because the new name sounds a lot more logical. But it seems that `canBeFocused` will be `true` when the items are in fact **NOT** focusable, right? So maybe it would make sense to either

- change the condition to check for a positive value, instead of a negative one and leave the name as `canBeFocused`
- leave the condition as it is, and rename to something like `itemsNotFocusable`

Reply:
I forgot to switch the logic :/